### PR TITLE
[SW-472] Upgrade to Gradle 4.0

### PR DESCRIPTION
**Pull request by jakubhava**
- Also resolve several ScalaDoc warnings

### CODE REVIEW

File: gradle/scriptsTest.gradle [hunk 1]
```diff
@@ -28,7 +28,8 @@ configurations {
 task scriptsTest(type: Test) {
     group = 'Verification'
     description = 'Runs all scripts tests'
-    testClassesDir = sourceSets.scriptsTest.output.classesDir
+
+    testClassesDirs = sourceSets.scriptsTest.output.classesDirs
```
Comment (jakubhava) [inline gradle/scriptsTest.gradle]: note - it doesn't mean we don't generate javadoc for java classes, the scaladoc will just doesn't look in the empty java output directory since we are using scala output directoru

File: gradle/scala.gradle [hunk 2]
```diff
@@ -91,6 +99,11 @@ export H2O_EXTENDED_JAR=`./gradlew -q extendJar -PdownloadH2O`
     }
 }
 
+// Override ScalaDoc classpath to look only for classes in the scala output dir preventing several warnings for
+// missing elements since we don't use java source dirs
+scaladoc {
```
Comment (jakubhava) [inline gradle/scala.gradle]: note - it doesn't mean we don't generate javadoc for java classes, the scaladoc will just doesn't look in the empty java output directory since we are using scala output directory

File: gradle/itest.gradle [hunk 3]
```diff
@@ -30,7 +30,7 @@ configurations {
 task integTest(type: Test) {
     group = 'Verification'
     description = 'Runs all integration tests'
-    testClassesDir = sourceSets.integTest.output.classesDir
+    testClassesDirs = sourceSets.integTest.output.classesDirs
```
Comment (mmalohlava) [inline gradle/itest.gradle]: Why this change?

**Comment by jakubhava** [comment]
Gradle deprecation warnings starting with Gradle 4.0
